File: app/src/main/java/com/boardgamegeek/service/CollectionCommentUploadTask.java
package com.boardgamegeek.service;

import android.support.annotation.NonNull;

import com.boardgamegeek.provider.BggContract.Collection;
import com.boardgamegeek.service.model.CollectionItem;

import okhttp3.OkHttpClient;

public class CollectionCommentUploadTask extends CollectionTextUploadTask {
	public CollectionCommentUploadTask(OkHttpClient client) {
		super(client);
	}

	@NonNull
	protected String getTextColumn() {
		return Collection.COMMENT;
	}

	@Override
	public String getTimestampColumn() {
		return Collection.COMMENT_DIRTY_TIMESTAMP;
	}

	@NonNull
	protected String getFieldName() {
		return "comment";
	}

	protected String getValue(CollectionItem collectionItem) {
		return collectionItem.getComment();
	}

	@Override
	public boolean isDirty() {
		return collectionItem.getCommentTimestamp() > 0;
	}
}

File: app/src/main/java/com/boardgamegeek/service/CollectionCommentUploadTask.kt
package com.boardgamegeek.service

import com.boardgamegeek.provider.BggContract.Collection
import okhttp3.OkHttpClient

class CollectionCommentUploadTask(client: OkHttpClient) : CollectionTextUploadTask(client) {
    override val timestampColumn = Collection.COMMENT_DIRTY_TIMESTAMP

    override val fieldName = "comment"

    override fun getValue() = collectionItem.comment.orEmpty()

    override val isDirty = collectionItem.commentTimestamp > 0
}
